AUG:
    RESIZE_METHOD: 'stepscaling' # choice unpadding rangescaling and stepscaling
    FIX_RESIZE_SIZE: [720, 720] # (width, height), for unpadding
    INF_RESIZE_VALUE: 500  # for rangescaling
    MAX_RESIZE_VALUE: 600  # for rangescaling
    MIN_RESIZE_VALUE: 400  # for rangescaling
    MAX_SCALE_FACTOR: 2.0  # for stepscaling
    MIN_SCALE_FACTOR: 0.75  # for stepscaling
    SCALE_STEP_SIZE: 0.25  # for stepscaling
    TRAIN_CROP_SIZE: [512, 256]  # crop size for training
    EVAL_CROP_SIZE: [512, 256]  # crop size for evaluating
    CROP_PAD_SIZE: 32
    MIRROR: True
    FLIP: False
    FLIP_RATIO: 0.5
    RICH_CROP:
        ENABLE: False
        BLUR: True
        BLUR_RATIO: 0.2
        MAX_ROTATION: 15
        MIN_AREA_RATIO: 0.5
        ASPECT_RATIO: 0.5
        BRIGHTNESS_JITTER_RATIO: 0.5
        CONTRAST_JITTER_RATIO: 0.5
        SATURATION_JITTER_RATIO: 0.5
DATASET:
    DATA_DIR: '/home/st/'
    IMAGE_TYPE: 'rgb'  # choice rgb or rgba
    NUM_CLASSES: 2
    TEST_FILE_LIST: 'REPO_ROOT_PATH/data/training_data_example/test.txt'
    TRAIN_FILE_LIST: 'REPO_ROOT_PATH/data/training_data_example/train.txt'
    VAL_FILE_LIST: 'REPO_ROOT_PATH/data/training_data_example/val.txt'
    IGNORE_INDEX: 255
    PADDING_VALUE: [127.5, 127.5, 127.5]
    MEAN_VALUE: [0.5, 0.5, 0.5]
    STD_VALUE: [0.5, 0.5, 0.5]
    CPU_MULTI_PROCESS_NUMS: 8
FREEZE:
    MODEL_FILENAME: 'model'
    PARAMS_FILENAME: 'params'
MODEL:
    MODEL_NAME: 'lanenet'
    FRONT_END: 'bisenetv2'  #vgg
    EMBEDDING_FEATS_DIMS: 4
    BISENETV2:
        GE_EXPAND_RATIO: 6
        SEMANTIC_CHANNEL_LAMBDA: 0.25
        SEGHEAD_CHANNEL_EXPAND_RATIO: 2
TEST:
    TEST_MODEL: 'model/cityscapes/final'
TRAIN:
    MODEL_SAVE_DIR: 'model/tusimple/'
    TBOARD_SAVE_DIR: 'tboard/tusimple/'
    MODEL_PARAMS_CONFIG_FILE_NAME: "model_train_config.json"
    RESTORE_FROM_SNAPSHOT:
        ENABLE: False
        SNAPSHOT_PATH: ''
    SNAPSHOT_EPOCH: 8
    BATCH_SIZE: 32
    VAL_BATCH_SIZE: 4
    EPOCH_NUMS: 905
    WARM_UP:
        ENABLE: True
        EPOCH_NUMS: 8
    FREEZE_BN:
        ENABLE: False
    COMPUTE_MIOU:
        ENABLE: True
        EPOCH: 1
    MULTI_GPU:
        ENABLE: True
        GPU_DEVICES: ['0', '1']
        CHIEF_DEVICE_INDEX: 0
SOLVER:
    LR: 0.001
    LR_POLICY: 'poly'
    LR_POLYNOMIAL_POWER: 0.9
    OPTIMIZER: 'sgd'
    MOMENTUM: 0.9
    WEIGHT_DECAY: 0.0005
    MOVING_AVE_DECAY: 0.9995
    LOSS_TYPE: 'cross_entropy'
GPU:
    GPU_MEMORY_FRACTION: 0.9
    TF_ALLOW_GROWTH: True
POSTPROCESS:
    MIN_AREA_THRESHOLD: 100
    DBSCAN_EPS: 0.35  #0.35
    DBSCAN_MIN_SAMPLES: 500 #1000

    # 自定义的聚类的一些参数
    IPM_DIVIDED_NUMS: 30  # 降采样时分成多少份
    VALID_DISTANCE: 125  #只取近处的结果，即pixel_y >...
    MY_CLUSTER_EPS: 60
    MY_CLUSTER_MIN_SAMPLES: 3
LOG:
    SAVE_DIR: './log'
    LEVEL: INFO

# 实测中的参数
ACTUAL:
    SRC_IMG_W: 1280
    SRC_IMG_H: 720
    DST_IMG_W: 512
    DST_IMG_H: 256
    # 2\4号相机的视场角的一半，单位弧度
    alpha_u: 0.908
    alpha_v: 0.750
    # 相机外参矩阵
    cameraMatrix_R: [0.0, 0.0, 0.0]  # 俯仰、航向、横滚
    cameraMatrix_T: [[0.0], [-1.05], [0.0]]  # X,Y,Z，Y垂直地面
    # 左目相机，4号内参、畸变
    cameraMatrix_K_4: [[1323.27, 0., 613.63],
                       [0., 1322.63, 304.68],
                       [0.0,     0.0,     1.0]]
    distCoeffs_4: [-0.48, 0.14 , 0.0, 0.0]
    # 右目相机，2号内参、畸变
    cameraMatrix_K_2: [[7.1900138539858858e+02, 0., 3.4779585424459759e+02],
                       [0., 8.3026888657248708e+02, 2.1885128384186845e+02],
                       [0.0,     0.0,     1.0]]
    distCoeffs_2: [-5.4931305852871426e-01, 5.8373850086066925e-01, 0.0, 0.0]
    # 仿真相机
    cameraMatrix_K_emulation: [[1312.961, 0.0,       640.00 ],
                               [0.0,      1312.961,  360.00],
                               [0.0,      0.0,       1.0]]
    cameraMatrix_T_emulation: [0.0, -1.165, 0.0]
    # 投影变换，路面范围
    X_min: -1.5
    X_max: 1.5
    Z_min: 3.5
    
    Z_max: 12.0
    # 投影变换后对应的像素点
    dst_pixels: [[175, 230], [175, 10], [375, 10], [375, 230]]


    # 左目相机，4号去畸变后的内参
    undis_cameraMatrix_K_4: [[1.2315846146586534e+03, 0., 5.7774543762207031e+02],
                             [0., 1.2315846146586534e+03, 2.7790315628051758e+02],
                            [0.0,     0.0,     1.0]]
    undis_distCoeffs_4: [0.0, 0.0, 0.0, 0.0]
